docs/StyleGuides.md

```diff
@@ -0,0 +1,48 @@
+## Code style
+
+Some code guidelines to keep in mind when contributing to laminar or amethyst-networking
+1. Comments
+    - Comment all code you’ve added. Things you should comment: types, methods/functions public fields of structs. 
+    - Calculations should be documented. Whether it would be in a PR or code. But it must be clear what is done.
+    - Public things should get docstring comments using ///. Non-public things may use // comments. 
+    - Keep comments small. 
+    - Don’t create unnecessary comments. They must add value.
+    - Comments should explain the “why” not the “what”.
+2. Hard Coding
+    - Don't hard code values anywhere.
+    - Use the ‘NetworkConfig’ type for common network settings, use consts or parameter input.
+    - Use of lazy_static is acceptable but first make sure you can’t fix the issue in other ways. 
+3. Code markup
+    - Keep files small. Better have small files with small pieces of logic than having one file with 1000 lines of logic with multiple types/structs etc. Note that I speak of logic, tests not included.
+    - Use ‘rust fmt’ newest version take note to open new PR when doing this. 
+Don’t run ‘rust fmt’, ‘cargo fix’ during an normal PR because this causes huge PR’s which are not possible to review properly.
+    - Absolutely no panics/unwraps in the logic. 
+      Follow the error handling pattern elsewhere in the code. 
+      You may have panics/unwraps in test code.
+ 
+# Library Import Order
+The following order of imports are used in the library. All imports should follow these patterns.
+
+**lib.rs file structure**
+1. macro extern crates.
+1. extern crates.
+1. private modules.
+1. public modules.
+1. private crate usings.
+1. public crate usings.
+1. private stdlib usings.
```

Comment:
Can we do this consistently? The common order is

1. std imports
2. extern crate imports (third party)
3. extern crate imports (workspace)
4. local imports

Reply:
I have removed it since we also do not maintain a common order yet. We'll move it to another issue. We had it discussed in a google doc about the order. But we'll need to revisit it since @LucioFranco also didn't like it.